crates/air_r_formatter/src/r/auxiliary/root.rs

```diff
@@ -23,4 +40,58 @@ impl FormatNodeRule<RRoot> for FormatRRoot {
             ]
         )
     }
+
+    fn is_suppressed(&self, node: &RRoot, f: &RFormatter) -> bool {
+        let Some(child) = node.expressions().first() else {
+            return false;
+        };
+
+        let comments = f.context().comments();
+
+        // Only consider very first comment in the file
+        let Some(comment) = comments.leading_comments(child.syntax()).first() else {
+            return false;
+        };
+
+        Self::is_suppression_comment(comment.piece().text())
+    }
+
+    // The default handling of suppressed nodes in `FormatNodeRule::fmt()`
+    // does not work out well for `RRoot` because it doesn't format
+    // trivia verbatim. So we use a custom implementation instead.
+    fn fmt_suppressed(&self, node: &RRoot, f: &mut RFormatter) -> FormatResult<()> {
+        let RRootFields { bom_token, .. } = node.as_fields();
+        write!(f, [bom_token.format()])?;
+
+        // Mark everything as visited so the formatter doesn't panic
+        // thinking we've missed elements. We're going to print each element
+        // manually with `dynamic_text()`.
+        mark_visited(node.syntax(), f);
+
+        // `node.text()` trims trivia so we get the raw text on the syntax
+        // node instead
+        let text: String = node.syntax().text().into();
```

Comment:
I wonder how expensive this folding of individual text pieces ends up being. Like, for very large generated files it does kind of stink that we have to waste time and actually do work on that file to "format" it, vs just recognizing the file as generated and not touching it whatsoever

Reply:
We can add an optimisation to the CLI by handling skip-file comments even before parsing happens. We've decided to do that only if it proves necessary though.